app/models/nic/interface.rb

```diff
@@ -23,7 +23,10 @@ class Interface < Base
     delegate :network, :to => :subnet
 
     def vlanid
-      self.tag.blank? ? self.subnet.vlanid : self.tag
+      return self.tag unless self.tag.blank?
+      return self.subnet.vlanid if self.subnet
```

Comment:
I think the condition should also verify that `subnet.vlanid.present?`, otherwise if interface is attached to ipv4 subnet without vlan, ipv6 subnet vlan id can't win?